### ISSUE TITLE
download filename can't support utf-8

### ISSUE THREAD

**Issue by lmtoo** [paulcwarren/spring-content#81]
if original filename is 表单ID及字段.xlsx ,then downloaded filename is ??ID???.xlsx, response header  Content-Disposition is `form-data; name="attachment"; filename="??ID???.xlsx"`

**Comment by paulcwarren** [comment]
Thanks.  We can investigate this one.

**Comment by paulcwarren** [comment]
@lmtoo we were missing coverage for I18Ned filenames so I added that but the tests immediately went green.  I was expecting them to fail.  So, either I have misinterpreted the issue, or it is more of a configuration thing.  

The tests I added are [here](https://github.com/paulcwarren/spring-content/blob/ab6bae1e39d5dbea1ca449010d45df1c83ca99ae/spring-content-rest/src/test/java/internal/org/springframework/content/rest/controllers/Content.java#L81) and [here](https://github.com/paulcwarren/spring-content/blob/ab6bae1e39d5dbea1ca449010d45df1c83ca99ae/spring-content-rest/src/test/java/internal/org/springframework/content/rest/controllers/Content.java#L108).  The build that went green with these tests is [here](https://travis-ci.org/paulcwarren/spring-content/builds/641246264).

Would you mind checking those two tests and see if I have the issue understood correctly.  Thanks.

**Comment by paulcwarren** [comment]
I will test some additional combinations my end too in using [spring-docs](https://github.com/paulcwarren/spring-docs).

**Comment by lmtoo** [comment]
MockMvc doesn't running in a servlet container , http header value maybe encoded to ISO-8859-1  by the servlet container , see [https://tools.ietf.org/html/rfc2616#section-2.2](url)

**Comment by lmtoo** [comment]
the assert should be `
assertThat(response.getHeader("Content-Disposition"), equalTo(URLEncoder.encode("表单ID及字段.txt", "UTF-8")));
`

**Comment by paulcwarren** [comment]
Thanks.  Digging into that theory [a bit further](https://stackoverflow.com/a/6745788/5609837) it looks like we should actually be asserting this:

```
assertThat(response.getHeader("Content-Disposition"), containsString("filename*=UTF-8''" + URLEncoder.encode("表单ID及字段.txt")));
```

This seems to work for me when testing using our spring-docs application (and chrome browser).  Would that work for you?

NB: I committed this change so its available via 1.0.0.M6-SNAPSHOT if you want to try it out.  I'll leave this open for the time being.

**Comment by paulcwarren** [comment]
Nothing heard back @lmtoo so based on my testing and reading I assume this is fixed.  Closing.  Feel free to reopen if you still have an issue and we can investigate further.
